### PR TITLE
6329-setUp-template-proposed-by-calypso-could-contain-super-call

### PR DESCRIPTION
**Pull request by olekscode**
Fixing #6329

### CODE REVIEW

**Comment by MarcusDenker** [comment]
failing test not related

**Comment by dionisiydk** [comment]
@olekscode One of the purposes of setUp tab was to assist with new setUp implementation. Due to simplistic approach the missing part was to provide a good template instead of raw empty TestCase>>setUp. 
Is it intentional to remove this feature?

**Comment by olekscode** [comment]
@dionisiydk yes, it was intentional. What we had was a `setUp` tab that always contained the code from a superclass. In most cases, this would be an empty method from TestCase, in some cases, it would be a `setUp` method from a parent test.

Having an empty template for a method is not harmful, but also not very useful. And there was no easy fix that would improve the template by adding `super setUp` to it, because it would require adding this line to `TestCase` which does not have any superclass that implements `setUp`.

What's worse is that every time you modify the proposed "template" and try to save it, you get a pop-up asking you whether you want to add the new method to your class or to superclass (TestCase). And some people find it annoying because "of course I want to add the method to the class which I'm currently editing".

Having proper templates for `setUp` and `tearDown` would indeed be nice, but it would require a lot of added effort, because currently we do not have the mechanism for that (showing the superclass implementation might not be good, as I explained above).  And since having those two templates is not that important, we decided that it was better to remove this functionality for now and only show the `setUp` and `tearDown` tabs for classes that implement those methods themselves.

**Comment by olekscode** [comment]
@MarcusDenker do you know what can be causing this test to fail?
Because indeed it seems unrelated

https://ci.inria.fr/pharo-ci-jenkins2/blue/organizations/jenkins/Test%20pending%20pull%20request%20and%20branch%20Pipeline/detail/PR-6704/1/tests

<blockquote><div><strong><a href="https://ci.inria.fr/pharo-ci-jenkins2/blue/organizations/jenkins/Test%20pending%20pull%20request%20and%20branch%20Pipeline/detail/PR-6704/1/tests">Jenkins</a></strong></div></blockquote>

File: src/Calypso-SystemPlugins-SUnit-Browser/ClyTestSetUpEditorTool.class.st [hunk 1]
```diff
@@ -30,7 +30,7 @@ ClyTestSetUpEditorTool class >> shouldBeActivatedInContext: aBrowserContext [
 	
 	(aBrowserContext lastSelectedItem hasProperty: ClyTestResultProperty) ifFalse: [ ^false ].
 	
-	(aBrowserContext lastSelectedClass lookupSelector: #setUp) ifNil: [ ^false ].
+	(aBrowserContext lastSelectedClass selectors includes: #setUp) ifFalse: [ ^false ].
```
Comment (dionisiydk) [inline src/Calypso-SystemPlugins-SUnit-Browser/ClyTestSetUpEditorTool.class.st]:
```suggestion
	(aBrowserContext lastSelectedClass includesSelector: #setUp) ifFalse: [ ^false ].
```

**Comment by dionisiydk** [comment]
> Having proper templates for `setUp` and `tearDown` would indeed be nice, but it would require a lot of added effort, because currently we do not have the mechanism for that (showing the superclass implementation might not be good, as I explained above).

I do not get where this "a lot of effort required" is coming from. But challenge accepted :)). 
I created PR with improvement instead of feature removal: #6717 . You can see that it was easy to do.

**Comment by MarcusDenker** [comment]
#6717 has been merged

**Comment by olekscode** [comment]
Nice! 😄 
Indeed, I might have overestimated the effort that is needed.

**Comment by MarcusDenker** [comment]
So this can be closed now?

**Comment by Ducasse** [comment]
oleksandr should we close this issue?

**Comment by olekscode** [comment]
Yes

**Pull request closed without merging**